Assets/_BForBoss/_Weapons/Scripts/ScriptableObjects/Melee/MeleeScriptableObject.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Perigon.Weapons
+{
+    public interface IMeleeProperties
+    {
+        float Damage { get; }
+        float AttackCoolDown { get; }
+
+        int OverlapCapsule(Vector3 position, Vector3 forwardDirection, ref Collider[] buffer);
+    }
+    
+    [CreateAssetMenu(fileName = "MeleeProperties", menuName = "PerigonGames/Melee", order = 1)]
+    public class MeleeScriptableObject : ScriptableObject, IMeleeProperties
+    {
+        [SerializeField] private float _damage = 5f;
+        [SerializeField] private float _coolDown = 0.5f;
+        
+        [field: SerializeField] public float Range { get; set; } = 1f;
+        [field: SerializeField] public float Height { get; set; } = 2f;
+
+        public float Damage => _damage;
+        public float AttackCoolDown => _coolDown;
+
+        public int OverlapCapsule(Vector3 position, Vector3 forwardDirection, ref Collider[] buffer)
```

Comment:
Its okay now, I have a rough approximate for the gizmos with two spheres that works better than it sounds. I also found some code on the forums for drawing a capsule with the handles api, so if we decide to add editor tooling it would be pretty easy to swap out for a proper capsule